pycalphad/io/cs_dat.py

```diff
@@ -1159,6 +1162,999 @@ def parse_cs_dat(instring):
+            # Get excess magnetic parameters
+            for tc in tcs:
+                tc_value = tc['parameter']
+                tc_constituents = tc['constituent_array']
+                tcs.remove(tc)
+                bmagn_value = 0
+                for bmagn in bmagns:
+                    # Look for matching bmagn
+                    if bmagn['constituent_array'] != tc_constituents:
+                        continue
+                    bmagn_value = bmagn['parameter']
+                    bmagns.remove(bmagn)
+                    break
+
+                # Index calculation depends on model
+                if   phase_model == 'RKMPM':
+                    # Get indices of participating constituents in phase (order of printed endmembers)
+                    indices = []
+                    for species in constituent_set:
+                        name = ''
+                        for element in species.constituents:
+                            # Add element name to endmember name if not vacancy
+                            if element.capitalize() != 'Va':
+                                name += element.capitalize()
+                                stoich = species.constituents[element] * weight
+                                # Add stoichiometric factor to name if not 1
+                                if stoich != 1:
+                                    if stoich % 1 == 0:
+                                        name += f'{int(stoich)}'
+                                    else:
+                                        name += f'{stoich:.2g}'
+                        try:
+                            indices.append(1 + endmember_names.index(name))
+                        except ValueError:
+                            print(f'Endmember {name} not found in phase {phase_name}')
+                            print(f'List of known endmembers: {endmember_names}')
+                            print('Write aborted')
+                            return
+                elif phase_model == 'SUBLM':
+                    # Get indices of participating constituents in phase (order of printed endmembers)
+                    indices = []
+                    for sublattice in tc_constituents:
+                        for species in sublattice:
+                            try:
+                                indices.append(1 + flat_constituents.index(species.name))
+                            except ValueError:
+                                print(f'Can\'t find constituent {species.name}')
+                # Now write excess magnetic terms for current constituent_set
+                output += f'{len(indices):4}\n'
+                # TODO: Get order properly if possible
+                order = 1
+                output += f'{"".join([f"{ind:4}" for ind in indices])}{order:4}\n'
+                # Check if magnetic parameters are provided as piecewise
+                if type(tc_value) == type(Piecewise([0,True])):
```

Comment:
Is type checking what we want to do here, or should it just be value comparison?

Reply:
Deliberately checking the type. Usually these are floats, but I found some come in as Piecewise instead.

The Piecewise terms in `alfe_sei.TDB` are equivalent to floats (so I assume this will happen in other cases too). Looks like a good place to use the user-selected `incompatibility` handler.